src/mongo/MongoScrapbook.ts

```diff
@@ -348,35 +348,55 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 		}
 		let argumentContextArray: mongoParser.ArgumentContext[] = filterType(argumentsToken.children, mongoParser.ArgumentContext);
 
-		let functionMap = { "ObjectId": this.objectIdToObject, "ISODate": this.dateToObject, "Date": this.dateToObject };
+		let functionMap = { "ObjectId": this.objectIdToObject, "ISODate": this.isodateToObject, "Date": this.dateToObject };
 		if (argumentContextArray.length > 1) {
 			let err: IParsedError = parseError(`Too many arguments. Expecting 0 or 1 argument(s) to ${constructorCall}`);
 			this.addErrorToCommand(err, ctx);
 			return {};
 		}
+
 		if (constructorCall.text in functionMap) {
```

Comment:
This `functionMap` thing is really weird to me. I feel like it'll make our lives easier if you just do a simple `switch` like this:

```typescript
const tokenText: string | undefined = argumentContextArray.length ? argumentContextArray[0].text : undefined;
switch (constructorCall.text) {
	case 'ObjectId':
		return this.objectIdToObject(ctx, tokenText);
	case 'ISODate':
		return this.dateToObject(ctx, tokenText, true /* isIsoDate */);
	case 'Date':
		return this.dateToObject(ctx, tokenText);
	default:
		// ...
}
```